presto-main/src/test/java/io/prestosql/sql/planner/TestTableScanRedirectionWithPushdown.java

```diff
@@ -201,6 +211,25 @@ public void testRedirectAfterColumnPruningOnPushedDownPredicate()
         }
     }
 
+    @Test
+    public void testPredicateTypeMismatch()
+    {
+        try (LocalQueryRunner queryRunner = createLocalQueryRunner(
+                getMockApplyRedirectAfterPredicatePushdown(typeMismatchedRedirectionMappingBC, Optional.of(ImmutableSet.of(sourceColumnHandleB))),
+                Optional.of(this::mockApplyProjection),
+                Optional.of(getMockApplyFilter(ImmutableSet.of(sourceColumnHandleC))))) {
+            // After 'source_col_c = 1' is pushed into source table scan, it's possible for 'source_col_c' table scan assignment to be pruned
+            // Redirection results in Project('dest_col_b') -> Filter('dest_col_c = 1') -> TableScan for such case
+            // but dest_col_a has mismatched type compared to source domain
```

Comment:
What's `dest_col_a` in this example?

Reply:
`dest_col_a` is integer

Reply:
Sorry, I meant what role does it play since it seems unrelated to the example given. But then I noticed you changed this to say `dest_col_c` in another commit.

Reply:
`source_col_c` gets redirected to `dest_col_a`, but `dest_col_a` has different type (int) than `source_col_c` (varchar)